### PR TITLE
Remove user from Tenant_Person table.

### PR DESCRIPTION
**Pull request by dbgoaco**
Bug fix: [DP-751](https://noticingsystem.atlassian.net/jira/software/projects/DP/boards/22?selectedIssue=DP-751)

### CODE REVIEW

File: Services/CO.CDP.Organisation.WebApi/UseCase/RemovePersonFromOrganisationUseCase.cs [hunk 1]
```diff
@@ -17,9 +17,14 @@ public async Task<bool> Execute((Guid organisationId, RemovePersonFromOrganisati
         if (person == null) return await Task.FromResult(false);
 
         var organisationPerson = organisation.OrganisationPersons.FindLast(op => op.PersonId == person.Id);
-
+
```
Comment (jakzal) [inline Services/CO.CDP.Organisation.WebApi/UseCase/RemovePersonFromOrganisationUseCase.cs:20]: Unnecessary spaces.

File: Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs [hunk 2]
```diff
@@ -38,6 +38,13 @@ public async Task<IEnumerable<OrganisationPerson>> FindOrganisationPersons(Guid
         return await context.Set<OrganisationPerson>().FirstOrDefaultAsync(o => o.Organisation.Guid == organisationId && o.Person.Guid == personId);
     }
 
+    public async Task<Person?> FindPersonWithTenant(Guid personId)
+    {
+        return await context.Persons
+            .Include(p => p.Tenants)
+            .FirstOrDefaultAsync(o => o.Guid == personId);
+    }
```
Comment (jakzal) [inline Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs:46]: Missing test case.

File: Services/CO.CDP.OrganisationInformation.Persistence/IOrganisationRepository.cs [hunk 3]
```diff
@@ -15,6 +15,8 @@ public interface IOrganisationRepository : IDisposable
 
     public Task<OrganisationPerson?> FindOrganisationPerson(Guid organisationId, string userUrn);
 
+    public Task<Person?> FindPersonWithTenant(Guid personId);
```
Comment (jakzal) [inline Services/CO.CDP.OrganisationInformation.Persistence/IOrganisationRepository.cs:18]: OrganisationRepository doesn't seem like the right place to look for persons.

File: Services/CO.CDP.Organisation.WebApi/UseCase/RemovePersonFromOrganisationUseCase.cs (hunk 1 above)
Comment (dbgoaco) [reply Services/CO.CDP.Organisation.WebApi/UseCase/RemovePersonFromOrganisationUseCase.cs:20]: Done.

File: Services/CO.CDP.OrganisationInformation.Persistence/IOrganisationRepository.cs (hunk 3 above)
Comment (dbgoaco) [reply Services/CO.CDP.OrganisationInformation.Persistence/IOrganisationRepository.cs:18]: I thought the same yesterday evening. Moved to IPersonRepository.

File: Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs (hunk 2 above)
Comment (dbgoaco) [reply Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs:46]: Added unit tests to DatabasePersonRepository.cs

**Review by maciej-goaco — APPROVED** [review]

File: Services/CO.CDP.OrganisationInformation.Persistence.Tests/DatabasePersonRepositoryTest.cs [hunk 4]
```diff
@@ -103,6 +103,49 @@ public async Task ItCorrectlyAssociatesPersonWithOrganisation()
...
+        var organisation = GivenOrganisation(guid: Guid.NewGuid(), name: "TheOrganisation_Tenant1");
+
+        person.Tenants.Add(tenant);
+        person.Organisations.Add(organisation);
+        repository.Save(person);
+
+        var result = await repository.FindPersonWithTenant(person.Guid);
+
+        result.Should().NotBeNull();
+        result.Guid.Should().Be(person.Guid);
```
Comment (andymantell) [inline Services/CO.CDP.OrganisationInformation.Persistence.Tests/DatabasePersonRepositoryTest.cs:121]: There are some static analysis warnings here

**Review by andymantell — CHANGES_REQUESTED** [review]

File: Services/CO.CDP.OrganisationInformation.Persistence.Tests/DatabasePersonRepositoryTest.cs (hunk 4 above)
Comment (dbgoaco) [reply Services/CO.CDP.OrganisationInformation.Persistence.Tests/DatabasePersonRepositoryTest.cs]: Done.

**Review by andymantell — CHANGES_REQUESTED** [review]
This doesn't work for me. If I add a user to an organisation, and then remove them again, the organisation still appears on their dashboard.

**Comment by dbgoaco** [comment]
> This doesn't work for me. If I add a user to an organisation, and then remove them again, the organisation still appears on their dashboard.

Previously worked on SHA-1: 810eb2e7b46c24d68a8766b44713fe6461a82491 suspect merge to main has broken it.

**Review by andymantell — APPROVED** [review]
Tested and working locally

**Pull request merged**